src/ecs.js

```diff
@@ -26,7 +26,8 @@ const matchEntity = (entity) => {
 };
 
 const ejectEntity = (entity) => {
-  Object.values(entity.components).forEach(
```

Comment:
Could it be better this way?
```javascript
for (const key in entity) {
    if (entity.hasOwnProperty(key)) {
        const component = entity[key];
        component && component.destructor && component.destructor();
    }
}
```

Reply:
Looks  really good to me, but your eslint configuration does not seem to like it.

Actually I first tried for...of loops but also eslint did complain ;)

"  29:3   error  for..in loops iterate over the entire prototype chain, which is virtually never what you want. Use Object.{keys,values,entries}, and iterate over the resulting array  no-restricted-syntax
  30:16  error  Do not access Object.prototype method 'hasOwnProperty' from target object                                                                                              no-prototype-builtins
"

Reply:
Try this
```javascript
// eslint-disable-next-line no-restricted-syntax
for (const key in entity) {
    if (Object.prototype.hasOwnProperty.call(entity, key)) {
        const component = entity[key];
        component && component.destructor && component.destructor();
    }
}
```

Reply:
@kutuluk Sure, I have updated the pull request.